docs/designs/reference_link.md

```diff
@@ -0,0 +1,194 @@
+[\\]: # (TODO: correct the links once the corresponding docs have been added)
+# Reference Link
+Markdown provides a [syntax](https://spec.commonmark.org/0.28/#inline-link) to create hyperlinks.
+For example, the following syntax:
+
+```markdown
+[bing](http://www.bing.com)
+```
+or 
+```markdown
+[resource](../about.html)
+```
+
+Here the link could be either absolute url pointing to external resource(`www.bing.com` in the above example),
+or a relative path pointing to a local resource on the same machine (for example, `about.html`).
+
+DocFX also provide ways to link documents together, and we need follow some rules to make it easy to switch between platforms without friction.
+  - The relative path pointing to a local resource should be **case sensitive**.
+  - We support both forward slash `\` and back slash `/` in the link.
+  - We **don't** allow two files with same name but different casing in same repo.
+
+## Link to a local resource
+
+In DocFX, you can link to a local resource to:
+  - Create a hyperlink, like `[docfx](docs/design/tableofcontent.md)`
+  - Include a token, like `[!include[file name](subfolder/token.md)]`
```

Comment:
Does code snippet belong to this case?

[http://dotnet.github.io/docfx/spec/docfx_flavored_markdown.html?tabs=tabid-1%2Ctabid-a#code-snippet](http://dotnet.github.io/docfx/spec/docfx_flavored_markdown.html?tabs=tabid-1%252Ctabid-a#code-snippet)

Reply:
i think so, will add it

---
In reply to: [187233073](https://github.com/dotnet/docfx/pull/2752#discussion_r187233073) [](ancestors = 187233073)